### PR TITLE
Ionic thought creation

### PR DESCRIPTION
**Pull request by aljoh**
### Description: 
Adds function call to api to create thought

### CODE REVIEW

File: src/pages/thought_modal/thought_modal.ts [hunk 1]
```diff
@@ -1,28 +1,50 @@
...
+    this.presentToast(flash_msg)
+  }
+
   ionViewDidLoad() {
   }
 
+  createThought() {
+    this.thoughtsProvider.saveThought(this.thought)
+    .subscribe(data => this.thoughtCreate("Thought was successfully created."),
+              data => this.thoughtCreate(data.json().error[0]))
```
Comment (diraulo) [inline src/pages/thought_modal/thought_modal.ts]: ~~Why are we calling `this.thoughtCreate` again here?~~

~~This is supposed to be the `onError` where you handle any errors that occur while executing the request. I would expect to see something along the lines of~~

```javascript
.subscribe(
  data => doSomethingOnSuccess(),
  error => doSomethingWithError()
)
```

Perhaps find a better name for `thoughtCreate()` -- I only realized half way through my comment that you actually use that to display various messages to the user

Comment (diraulo) [reply src/pages/thought_modal/thought_modal.ts]: something like this is probably better

```javascript
.subscribe(
  data => {
    this.presentToast("Thought was successfully created.");
    this.closeModal();
  },
  error => this.presentToast(getErrorMessageFrom(error))
)
```

IMO You don't really need that `thoughtCreate()` method

**Review by diraulo — COMMENTED** [review]
Few comments for you..

File: src/pages/thought_modal/thought_modal.ts (hunk 1 above)
Comment (aljoh) [reply src/pages/thought_modal/thought_modal.ts]: Fixed : )

**Pull request merged**